src/com/google/enterprise/adaptor/database/DatabaseAdaptor.java

```diff
@@ -193,12 +193,20 @@ public void init(AdaptorContext context) throws Exception {
       log.config("adaptor runs in lister-only mode");
     }
 
-    modeOfOperation = cfg.getValue("db.modeOfOperation");
+    modeOfOperation = cfg.getValue("db.modeOfOperation").trim();
     if ("urlAndMetadataLister".equals(modeOfOperation) && encodeDocId) {
       String errmsg = "db.modeOfOperation of \"" + modeOfOperation
           + "\" requires docId.isUrl to be \"true\"";
       throw new InvalidConfigurationException(errmsg);
     }
+    String modeOfOperationValid[] = {"blobColumn", "filepathColumn",
```

Comment:
loadResponseGenerator validates modeOfOperation values by reflection (they are generally methods on ResponseGenerator, though apparently custom values are also permitted). You don't have to worry about this, you can just add a TODO for me, or ignore it, since I'm unlikely to forget.